src/PAModel/Utility/DirectoryHelper.cs

```diff
@@ -129,6 +132,20 @@ public bool FileExists(string subdir, string filename)
             string path = Path.Combine(_directory, subdir, filename);
             return File.Exists(path);
         }
+
+        /// <summary>
+        /// Returns true if it's either an empty directory or it contains CanvasManifest.json file.
+        /// </summary>
+        /// <returns></returns>
+        private bool IsValidForDelete(ErrorContainer errors)
```

Comment:
>bool IsValidForDelete [](http://example.com/codeflow?start=16&length=21)

nit: since this either returns true or Throws (never returns false), it could be:
void ValidateSafeToDelete(errors)